libsolidity/ast/AsmJsonConverter.cpp

```diff
@@ -0,0 +1,220 @@
+/*
+  This file is part of solidity.
+
+  solidity is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  solidity is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @date 2019
+ * @author julius <djudju@protonmail.com>
+ * Converts inline assembly AST to JSON format
+ */
+
+#include <libsolidity/ast/AsmJsonConverter.h>
+#include <libyul/AsmDataForward.h>
+
+using namespace std;
+
+namespace dev
+{
+namespace solidity
+{
+
+
+Json::Value AsmJsonConverter::addSourceLocation(Json::Value const& _node, langutil::SourceLocation _location) {
+	int length = -1;
+	if (_location.start >= 0 && _location.end >= 0)
+		length = _location.end - _location.start;
+	_node["src"] = to_string(_location.start) + ":" + to_string(length) + ":" + to_string(m_sourceIndex);
```

Comment:
Is this the format also used in the other json ast?

Reply:
yes, I copied it from there...

NB: It also is the only place where I had to introduce state in the AsmPrinter, to save the index of the source....